src/lib/utils/liquidityMining/index.ts

```diff
@@ -0,0 +1,68 @@
+import { differenceInWeeks } from 'date-fns';
+import { groupBy, mapValues, mergeWith, add } from 'lodash';
+import { bnum } from '..';
+
+import LiquidityMiningV2 from './LiquidityMiningV2.json';
+
+type LiquidityMiningSlot = Record<string, string>;
+
+type LiquidityMiningTierLevel = '1' | '2' | '3';
+
+type LiquidityMiningTier = {
+  BAL: number;
+  slots: LiquidityMiningSlot;
+};
+
+type LiquidityMiningWeek = {
+  start_timestamp: number;
+  end_timestamp: number;
+  tiers: Record<LiquidityMiningTierLevel, LiquidityMiningTier>;
+};
+
+type LiquidityMiningRewards = Record<string, number>;
+
+// Liquidity mining started on June 1, 2020
+function getCurrentLiquidityMiningWeek() {
+  return differenceInWeeks(new Date(), new Date(2020, 5, 1)) + 1;
+}
+
+function computeRewardsForTier(tier: LiquidityMiningTier) {
+  return mapValues(groupBy(tier.slots), slot => slot.length * tier.BAL);
+}
+
+export function computeAPYForPool(
+  rewards: number,
+  BALPrice: string,
+  totalLiquidity: string
+) {
+  return bnum(rewards)
+    .div(7)
+    .times(BALPrice)
+    .times(365)
+    .div(totalLiquidity)
+    .div(100)
+    .toNumber();
```

Comment:
@markusbkoch does this look right?

Reply:
I was checking the numbers on the UI and they all seem to be 1/100 of what they should. I think you need to remove line 43 here.